File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head
        while(curr):
            first = curr
            if(curr.next):
                second = curr.next
            else:
                break
            if(second):
                first.val, second.val = second.val, first.val
            curr = curr.next
            curr = curr.next
        return head

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* curr;
        curr = head;
        while(curr != NULL && curr->next != NULL)
        {
            ListNode* first = curr;
            ListNode* second = curr->next;
            int temp;
            temp = first->val;
            first->val = second->val;
            second->val = temp;
            curr = curr->next;
            curr = curr->next;
        }
        return head;
    }
};